flipt-python/flipt/authentication/__init__.py

```diff
@@ -17,3 +23,46 @@ def __init__(self, token: str) -> None:
 
     def authenticate(self, headers: dict[str, str]) -> None:
         headers["Authorization"] = f"JWT {self.token}"
+
+
+class KubernetesAuthentication(AuthenticationStrategy):
+    default_service_token_path = "/var/run/secrets/kubernetes.io/serviceaccount/token"  # noqa: S105
+
+    def __init__(self, token: str, service_account_token_path: str = default_service_token_path) -> None:
+        self.token = token
+        self.service_account_token_path = service_account_token_path
+        self.token_expiry = None
```

Comment:
@erka has correctly pointed out that we have hard-coded our URL to be one that works in our test suite, but needs to be configurable in the real world (and the test suite will set it appropriately.

I also noticed that we're passing in a token, that we won't need or know when we create this client. So instead lets change this parameter to be a string for the url.
```suggestion
class KubernetesAuthentication(AuthenticationStrategy):
    default_service_token_path = "/var/run/secrets/kubernetes.io/serviceaccount/token"  # noqa: S105

    def __init__(self, url: str, service_account_token_path: str = default_service_token_path) -> None:
        self.service_account_token_path = service_account_token_path
        self.url = url
        self.token = None
        self.token_expiry = None
```